synapse/replication/tcp/handler.py

```diff
@@ -663,6 +669,15 @@ def on_LOCK_RELEASED(
             cmd.instance_name, cmd.lock_name, cmd.lock_key
         )
 
+    async def on_NEW_ACTIVE_TASK(
+        self, conn: IReplicationConnection, cmd: NewActiveTaskCommand
+    ) -> None:
+        """Called when get a new NEW_ACTIVE_TASK command."""
+        if self._task_scheduler:
+            task = await self._task_scheduler.get_task(cmd.data)
+            if task and task.status == TaskStatus.ACTIVE:
+                await self._task_scheduler._launch_task(task)
```

Comment:
If this is command is only sent for new *active* task, why would its status not be active?

Reply:
It's just a sanity check, it shouldn't really happen indeed. I'll nuke it.